lib/transport/transport-factory-registry.c

```diff
@@ -50,17 +50,18 @@ TransportFactoryRegistry *transport_factory_registry_new(void)
   return instance;
 }
 
-void transport_factory_registry_free(TransportFactoryRegistry *self)
+void
+transport_factory_registry_free(TransportFactoryRegistry *self)
 {
   g_hash_table_unref(self->registry);
   g_free(self);
 }
 
-gboolean
+void
 transport_factory_registry_add(TransportFactoryRegistry *self, TransportFactory *factory)
 {
```

Comment:
and what if we’d create an insert method in compat that reports back result of insertion (lookup&insert?)
I didn’t cover the false case with unit tests, maybe we have to.

Reply:
Yeah, I would also check the return value of `transport_factory_registry_add()` in `multitransport_new()` and `multitransport_add_factory()`.

I'll add the compat function and fix all invocation in a separate PR.